subiquity/models/filesystem.py

```diff
@@ -785,10 +786,14 @@ def _can_be_boot_disk(self):
                     return True
                 else:
                     return self._partitions[0].flag == "bios_grub"
-            else:
-                flag = {Bootloader.UEFI: "boot", Bootloader.PREP: "prep"}[bl]
+            elif bl == Bootloader.UEFI:
+                for p in self._partitions:
```

Comment:
This can be simplified with an any statement to reduce usage of raw loops:

`return any(p.is_esp for p in self._partitions)`

Reply:
Yeah, that makes sense.